MediaBrowser.Model/Configuration/TrickplayOptions.cs

```diff
@@ -0,0 +1,60 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace MediaBrowser.Model.Configuration;
+
+/// <summary>
+/// Class TrickplayOptions.
+/// </summary>
+public class TrickplayOptions
+{
+    /// <summary>
+    /// Gets or sets a value indicating whether or not to use HW acceleration.
+    /// </summary>
+    public bool EnableHwAcceleration { get; set; } = false;
+
+    /// <summary>
+    /// Gets or sets the behavior used by trickplay provider on library scan/update.
+    /// </summary>
+    public TrickplayScanBehavior ScanBehavior { get; set; } = TrickplayScanBehavior.NonBlocking;
+
+    /// <summary>
+    /// Gets or sets the process priority for the ffmpeg process.
+    /// </summary>
+    public ProcessPriorityClass ProcessPriority { get; set; } = ProcessPriorityClass.BelowNormal;
+
+    /// <summary>
+    /// Gets or sets the interval, in ms, between each new trickplay image.
+    /// </summary>
+    public int Interval { get; set; } = 10000;
+
+    /// <summary>
+    /// Gets or sets the target width resolutions, in px, to generates preview images for.
+    /// </summary>
+    public int[] WidthResolutions { get; set; } = new[] { 320 };
+
+    /// <summary>
+    /// Gets or sets number of tile images to allow in X dimension.
+    /// </summary>
+    public int TileWidth { get; set; } = 10;
+
+    /// <summary>
+    /// Gets or sets number of tile images to allow in Y dimension.
+    /// </summary>
+    public int TileHeight { get; set; } = 10;
+
+    /// <summary>
+    /// Gets or sets the ffmpeg output quality level.
+    /// </summary>
+    public int Qscale { get; set; } = 4;
+
+    /// <summary>
+    /// Gets or sets the jpeg quality to use for image tiles.
+    /// </summary>
+    public int JpegQuality { get; set; } = 90;
+
+    /// <summary>
+    /// Gets or sets the number of threads to be used by ffmpeg.
+    /// </summary>
+    public int ProcessThreads { get; set; } = 0;
```

Comment:
What are your guys' thoughts on changing the default thread count to 1 for trickplay generation? I think it would be better to err on the side of caution and prevent people from turning this on and then not knowing why their cpu is at high usage while trickplay images are generating, which can potentially take days or weeks depending on library size and processing power.

Reply:
That is true, but on the other hand just 3 days ago i discovered that i can enable hardware acceleration for the jellyscrub task so now i enjoy some silence in my room again so that might be an better default value.

I think -1 (or use what is configured in transcoding) would be better. It is better to explain that the whole task is an resource intensive one then to explain why the new module is so much slower then the plugin one.

Reply:
The default previously was to used the same values as the rest of jellyfin, but the problem is that trickplay generation often runs continuously on larger libraries or slower hardware, leading to confusion when people see high resource usage. Even today a came across such a thread: [reddit.com/r/jellyfin/comments/13jlop9/ffmpeg_is_using_over_100_cpu/](https://www.reddit.com/r/jellyfin/comments/13jlop9/ffmpeg_is_using_over_100_cpu/).

Reply:
I see, however when i first run jellyscrub it took me well over a week with constant fan noise anyway. That came from the idea that the default settings where the most optimal so i assumed the whole process was just _slow_. I would assume that the default options are the fastest available so if that is no longer the case, the whole proccess would take months instead? that cannot be the solution, then a lot of people will complain that the scrub feature is not working because its so slow and i find that harder to explain to the avg user than a few days of constant usage.

But i see the dilemma here.

I would say, per default HW accelerated generation with max performance settings should be enabled.